app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDateTimePicker.kt

```diff
@@ -0,0 +1,226 @@
+import com.maxkeppeler.sheets.calendar.models.CalendarStyle
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.Locale
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun QuickFixDateTimePicker(
+    onDateTimeSelected: (LocalDate, LocalTime) -> Unit,
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    disableDates: List<LocalDate> = emptyList(),
+    timeViewTest: Boolean = true
+) {
+  var selectedDate by remember { mutableStateOf<LocalDate?>(null) }
+  var selectedTime by remember { mutableStateOf<LocalTime?>(null) }
+  var showDatePicker by remember { mutableStateOf(true) }
+  var showTimePicker by remember { mutableStateOf(false) }
+
+  if (showDatePicker) {
+    DatePickerDialog(
+        onDismissRequest = {
+          // Cancel the flow if dismissed
+          showDatePicker = false
+          onDismissRequest()
+        },
+        onDateSelected = { year, month, day -> selectedDate = LocalDate.of(year, month, day) },
+        onFinishedRequest = {
+          // Transition only if a date is selected
+          showDatePicker = false
+          showTimePicker = true
+        },
+        disableDates = disableDates)
+  }
+
+  if (showTimePicker) {
+    TimePickerDialog(
+        onDismissRequest = {
+          showTimePicker = false
+          onDismissRequest()
+        },
+        onTimeSelected = { hour, minute ->
+          selectedTime = LocalTime.of(hour, minute)
+          onDateTimeSelected(selectedDate!!, selectedTime!!)
+          showTimePicker = false
+        },
+        timeViewTest = timeViewTest)
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DatePickerDialog(
+    onDismissRequest: () -> Unit,
+    onDateSelected: (Int, Int, Int) -> Unit,
+    onFinishedRequest: () -> Unit,
+    disableDates: List<LocalDate> = emptyList(),
+) {
+  val useCaseState =
+      rememberUseCaseState(
+          visible = true,
+          onDismissRequest = {
+            // Handle cancel action explicitly
+            onDismissRequest()
+          },
+      )
+  MyAppTheme {
+    CalendarDialog(
+        state = useCaseState,
+        selection =
+            CalendarSelection.Date(
+                onSelectDate = { localDate ->
+                  // Trigger onDateSelected only when a date is selected
+                  onDateSelected(localDate.year, localDate.monthValue, localDate.dayOfMonth)
+                  onFinishedRequest()
+                },
+                selectedDate = LocalDate.now(),
+                onNegativeClick = { onDismissRequest() }),
+        config =
+            CalendarConfig(
+                locale = Locale.getDefault(),
+                style = CalendarStyle.WEEK,
+                cameraDate = LocalDate.now(),
+                yearSelection = true,
+                monthSelection = true,
+                disabledDates = disableDates),
+        header = Header.Default("Select Date"))
+  }
```

Comment:
Really like this, previously worked with this library and i implemented MyAppTheme to fix colors and typography, this is the exact intended usage, great job.